sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -201,6 +202,28 @@ public WritableColumnVector reserveDictionaryIds(int capacity) {
    */
   public abstract void putBooleans(int rowId, int count, boolean value);
 
+  /**
+   * Sets bits from [src[srcIndex], src[srcIndex + count]) to [rowId, rowId + count)
+   * src must contain bit-packed 8 booleans in the byte.
+   */
+  public void putBooleans(int rowId, int count, byte src, int srcIndex) {
```

Comment:
This as a public API looks a bit dangerous. `count` cannot be more than 7, but there is not assert on it. Can we add one?

Reply:
Thought about this too. Since this is hot path I think we can use `assert` but not `Preconditions` check. `putBytes` is also doing something very similar and there is no assertion there.

Reply:
Thanks, added!